### ISSUE TITLE
firebase_auth: ^0.8.0+1 crashes on Android

### ISSUE THREAD

**Issue by BitKnitting** [flutter/flutter#27156]
`firebase_auth: ^0.6.2+1`  worked.  
----------------------------------------------------------------
```
firebase_auth: ^0.8.0+1  errors during build.  Output from DEBUG CONSOLE:

Running "flutter packages get" in login_demo...                  0.7s
Launching lib/main.dart on Android SDK built for x86 in debug mode...
registerResGeneratingTask is deprecated, use registerGeneratedResFolders(FileCollection)
registerResGeneratingTask is deprecated, use registerGeneratedResFolders(FileCollection)
registerResGeneratingTask is deprecated, use registerGeneratedResFolders(FileCollection)
registerResGeneratingTask is deprecated, use registerGeneratedResFolders(FileCollection)
registerResGeneratingTask is deprecated, use registerGeneratedResFolders(FileCollection)
/Users/margaret 1/Flutter/flutter/.pub-cache/hosted/pub.dartlang.org/firebase_auth-0.8.0+1/android/src/main/java/io/flutter/plugins/firebaseauth/FirebaseAuthPlugin.java:9: error: cannot find symbol
import androidx.annotation.NonNull;
                          ^
  symbol:   class NonNull
  location: package androidx.annotation
/Users/margaret 1/Flutter/flutter/.pub-cache/hosted/pub.dartlang.org/firebase_auth-0.8.0+1/android/src/main/java/io/flutter/plugins/firebaseauth/FirebaseAuthPlugin.java:10: error: cannot find symbol
import androidx.annotation.Nullable;
                          ^
  symbol:   class Nullable
  location: package androidx.annotation
/Users/margaret 1/Flutter/flutter/.pub-cache/hosted/pub.dartlang.org/firebase_auth-0.8.0+1/android/src/main/java/io/flutter/plugins/firebaseauth/FirebaseAuthPlugin.java:638: error: cannot find symbol
  private void reportException(Result result, @Nullable Exception exception) {
                                               ^
  symbol:   class Nullable
  location: class FirebaseAuthPlugin
/Users/margaret 1/Flutter/flutter/.pub-cache/hosted/pub.dartlang.org/firebase_auth-0.8.0+1/android/src/main/java/io/flutter/plugins/firebaseauth/FirebaseAuthPlugin.java:550: error: cannot find symbol
    public void onComplete(@NonNull Task<AuthResult> task) {
                            ^
  symbol:   class NonNull
  location: class FirebaseAuthPlugin.SignInCompleteListener
/Users/margaret 1/Flutter/flutter/.pub-cache/hosted/pub.dartlang.org/firebase_auth-0.8.0+1/android/src/main/java/io/flutter/plugins/firebaseauth/FirebaseAuthPlugin.java:569: error: cannot find symbol
    public void onComplete(@NonNull Task<Void> task) {
                            ^
  symbol:   class NonNull
  location: class FirebaseAuthPlugin.TaskVoidCompleteListener
/Users/margaret 1/Flutter/flutter/.pub-cache/hosted/pub.dartlang.org/firebase_auth-0.8.0+1/android/src/main/java/io/flutter/plugins/firebaseauth/FirebaseAuthPlugin.java:587: error: cannot find symbol
    public void onComplete(@NonNull Task<SignInMethodQueryResult> task) {
                            ^
  symbol:   class NonNull
  location: class FirebaseAuthPlugin.GetSignInMethodsCompleteListener
/Users/margaret 1/Flutter/flutter/.pub-cache/hosted/pub.dartlang.org/firebase_auth-0.8.0+1/android/src/main/java/io/flutter/plugins/firebaseauth/FirebaseAuthPlugin.java:186: error: cannot find symbol
                      public void onComplete(@NonNull Task<AuthResult> task) {
                                              ^
  symbol: class NonNull
/Users/margaret 1/Flutter/flutter/.pub-cache/hosted/pub.dartlang.org/firebase_auth-0.8.0+1/android/src/main/java/io/flutter/plugins/firebaseauth/FirebaseAuthPlugin.java:445: error: cannot find symbol
              public void onComplete(@NonNull Task<GetTokenResult> task) {
                                      ^
  symbol: class NonNull
/Users/margaret 1/Flutter/flutter/.pub-cache/hosted/pub.dartlang.org/firebase_auth-0.8.0+1/android/src/main/java/io/flutter/plugins/firebaseauth/FirebaseAuthPlugin.java:499: error: cannot find symbol
          public void onAuthStateChanged(@NonNull FirebaseAuth firebaseAuth) {
                                          ^
  symbol: class NonNull
Note: /Users/margaret 1/Flutter/flutter/.pub-cache/hosted/pub.dartlang.org/firebase_auth-0.8.0+1/android/src/main/java/io/flutter/plugins/firebaseauth/FirebaseAuthPlugin.java uses unchecked or unsafe operations.
Note: Recompile with -Xlint:unchecked for details.
9 errors

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':firebase_auth:compileDebugJavaWithJavac'.
> Compilation failed; see the compiler error output for details.

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 9s
Gradle task assembleDebug failed with exit code 1
Exited (sigterm)
```

**Comment by wouterhardeman** [comment]
This is caused by: https://github.com/flutter/flutter/issues/27106 You will need to downgrade to 0.6.7  or update the `targetSdkVersion` in `android/app/build.gradle` to 28.

**Comment by BitKnitting** [comment]
Thank you.  I updated compleSdkVersion to 28 (android/app/build.gradle) get same error.

**Comment by wouterhardeman** [comment]
I would downgrade for now while the Flutter team is working on a solution.

**Comment by dev-vinicius** [comment]
Here i put this in file : android/gradle.properties:

android.useAndroidX=true
android.enableJetifier=true

and changed targetSdkVersion to 28 and it worked.

**Comment by BitKnitting** [comment]
Thank you.  Putting these two properties in gradle.properties worked for me also.  I still get: ```registerResGeneratingTask is deprecated, use registerGeneratedResFolders(FileCollection)``` which I am ignoring. It does look messy.

**Comment by yuryeuceda** [comment]
You can visit my answer to solve it.

[https://stackoverflow.com/questions/52409592/flutter-build-failed-with-an-exception/54394055#54394055](url)

**Comment by zoechi** [comment]
Duplicate of #27106

**Comment by alexandranb** [comment]
> This is caused by: #27106 You will need to downgrade to 0.6.7 or update the `compileSdkVersion` in `android/app/build.gradle` to 28.

What would I have to downgrade?

Thanks

**Comment by PreciousGabriel** [comment]
I added android.useAndroidX=true
android.enableJetifier=true  to android/gradle.properties and changed targetSdkVersion to 28 and it didn't work for me.

**Comment by PreciousGabriel** [comment]
> Here i put this in file: `android/gradle.properties:`
> 
> ```
> android.useAndroidX=true
> android.enableJetifier=true
> ```
> and changed `targetSdkVersion` to `28` and it worked.

I added 
android.useAndroidX=true
android.enableJetifier=true
to android/gradle.properties and changed targetSdkVersion to 28 and it didn't work for me.

**Comment by krngd2** [comment]
> > Here i put this in file: `android/gradle.properties:`
> > ```
> > android.useAndroidX=true
> > android.enableJetifier=true
> > ```
> > and changed `targetSdkVersion` to `28` and it worked.
> 
> I added
> 
> ```
> android.useAndroidX=true
> android.enableJetifier=true
> ```
> to `android/gradle.properties` and changed `targetSdkVersion` to `28` and it didn't work for me.

Yep, this isnt working for me too

**Comment by mb-mb** [comment]
isn't for me too

**Comment by zoechi** [comment]
You need to set `minSdkVersion` to `21` or add a dependency (don't remember) it's mentioned in the migration docs.

**Comment by mb-mb** [comment]
thank you @zoechi ... it works now.

**Comment by sivang** [comment]
could somebody please note here which migration is this talked about +docs to references the AndroidX migration and what it is?

Also having this, downgrading to firebase_auth 0.7.0 worked for me, but obviously sub optimal. is this an ABI incompatibility between minSdk 21 and targetSdk 28 ?

**Comment by Moswag** [comment]
addding these 2 to project/gradle.properties

android.useAndroidX=true
android.enableJetifier=true

it worked for me on firebase_auth: 0.8.2

**Comment by zolzaya** [comment]
I have a same problem :(

**Comment by mayurdhurpate** [comment]
@zolzaya  Please check this well-summarized answer by @mklim on this issue here: https://github.com/flutter/flutter/issues/27106#issuecomment-458366675

It says either one has to migrate to AndroidX to use the newer version of libraries (https://developer.android.com/jetpack/androidx/migrate) or pin the last pre-AndroidX supporting version of the library to prevent breaking changes.

**Comment by andrejvnior** [comment]
I am having the same issue. I spent a whole day looking for a solution e tried everything, but nothing works. If somebody has a good (and functional) tutorial about to login with Google using Firebase, I'll start over and try again, because I am about to get crazy with this. I was in love with Flutter + Firebase, but this really puts me down. I can't find a solution. 

I tried everything here in the topic, but nothing works for me.

**Comment by 5lineofcode** [comment]
Thank you, this works for me.

![image](https://user-images.githubusercontent.com/43805298/56400027-c0ffcd80-627b-11e9-8138-15aee5a2c288.png)

**Comment by Aidanvii7** [comment]
Alternatively:
1. open the android project in android studio
2. perform gradle sync
3. open the build.gradle file of the firebase_auth module
4. add the annotations dependency like so:
```
dependencies {
        api 'com.google.firebase:firebase-auth:16.0.5'
        api 'androidx.annotation:annotation:1.0.0' // << add this
    }
``` 
5. build parent flutter project again

**Comment by hariangr** [comment]
I fixed to problem by setting
android.useAndroidX=true
android.enableJetifier=true

AND
compileSdkVersion 28

As stated in https://flutter.dev/docs/development/packages-and-plugins/androidx-compatibility

> Under android {, make sure compileSdkVersion and targetSdkVersion are at least 28.

**Comment by GershonHerczeg** [comment]
```
android.useAndroidX=true
android.enableJetifier=true
```
Worked for me only after i added the above to this file as well: `firebase_auth/android/gradle.properties`

**Comment by analistacarlosh** [comment]
> Here i put this in file: `android/gradle.properties:`
> 
> ```
> android.useAndroidX=true
> android.enableJetifier=true
> ```
> 
> and changed `targetSdkVersion` to `28` and it worked.

Worked for me, Thank you so much.

**Comment by Mthandazo42** [comment]
I had the same issue i fixes it by adding 
android.useAndroidX=true
android.enableJetifier=true
inside the android/gradle.properties and upgraded my firebase_auth to current version which is 0.11 as well as change the compiled sdk from 27 to 28

**Comment by gcshri** [comment]
> ```
> android.useAndroidX=true
> android.enableJetifier=true
> ```
> 
> The above solution worked for me only after i added the above to this file as well: `firebase_auth/android/gradle.properties`

Thank you for this. Wish the flutter team had a doc related to common issues. AndroidX seems to be one of them.
